packages/entities/entities-gateway-services/src/components/GatewayServiceForm.vue

```diff
@@ -587,36 +686,174 @@ const changeCheckedGroup = () => {
+  form.errorMessages = []
+  initFieldDefaultValues()
+}
+
+const handleTrySampleApi = (): void => {
+  // Reset form fields before populating
+  initFieldDefaultValues()
+
+  form.fields.name = `${KongAirService.name} + ${new Date()
+    .toISOString() // Convert date to ISO string format (e.g., "2025-03-07T12:30:45.789Z")
+    .replace(/\D/g, '') // Remove all non-digit characters
+    .slice(0, 17)}` // Take the first 17 digits
+
+  switch (checkedGroup.value) {
+    case 'url':
+      // Construct the full URL
+      form.fields.url = `${KongAirService.protocol}://${KongAirService.host}${KongAirService.path}`
+      break
+
+    case 'protocol':
+      // Populate individual protocol-related fields
+      form.fields.host = KongAirService.host
+      form.fields.path = KongAirService.path
+      form.fields.protocol = KongAirService.protocol
+      form.fields.port = KongAirService.port
+      break
+
+    default:
+      form.fields.url = `${KongAirService.protocol}://${KongAirService.host}${KongAirService.path}`
+  }
+}
+
+const handleValidateFullUrl = useDebounceFn((): void => {
+  // reset the errors
+  resetFormFieldErrors()
+
+  if (form.fields.url.length) {
     try {
+      // try constructing a url
       const parsedUrl = new URL(form.fields.url)
-      // if the URL is parsed successfully, clear error message
-      form.errorMessage = ''
+      // url is valid
 
-      // remove the trailing colon appended to parsedUrl.protocol
+      // validate protocol
+      const protocolError = validateProtocol(parsedUrl.protocol, gatewayServiceProtocolItems)
+      if (protocolError) throw new Error(protocolError)
       form.fields.protocol = parsedUrl.protocol.slice(0, -1)
+
+      // validate hostname
+      const hostError = validateHost(parsedUrl.hostname)
+      if (hostError) throw new Error(hostError)
       form.fields.host = parsedUrl.hostname
+
+      // validate path
+      const pathError = validatePath(parsedUrl.pathname)
+      if (pathError) throw new Error(pathError)
       form.fields.path = parsedUrl.pathname
 
+      // validate port
+      const portError = validatePort(parsedUrl.port)
+      if (portError) throw new Error(portError)
       // extract port value from url and convert to number
       const portValue = Number(parsedUrl.port)
-
       hasExtractPortValue.value = !!portValue
       form.fields.port = portValue || getPort.getPortFromProtocol(form.fields.protocol)
 
-      form.errorMessage = ''
       emit('url-valid:success')
     } catch (error: any) {
-      form.errorMessage = t('errors.urlErrorMessage')
-      emit('url-valid:error', getMessageFromError(error))
+      emit('url-valid:error', error.message || 'URL validation failed')
+      form.formFieldErrors.url = error.message || 'URL validation failed'
     }
-  } else {
+  }
+}, 300)
+
+const handleValidateAdvancedFields = useDebounceFn((fieldId?: keyof FormFieldErrors) => {
+  // reset the errors
+  resetFormFieldErrors(fieldId ?? undefined)
```

Comment:
What's the purpose of `fieldId ?? undefined` here. Nullish will return the value even if it's falsey unless it is `null` or `undefined` but you are casting it to `undefined` anyway. Are you concerned it will be type `null`? `resetFormFieldErrors`'s parameter is optional (it will accept `undefined` and `null`), so there is no need for the nullish check here.